### PR TITLE
Add Taskflow

### PR DESCRIPTION
**Pull request by mpowelson**
Formally known as CPP Taskflow, this will be used in future Tesseract features.

### CODE REVIEW

**Comment by mpowelson** [comment]
It is worth noting that the ROS Industrial PPA has a debian of this package [here](https://launchpad.net/~ros-industrial/+archive/ubuntu/ppa)

This will be required by the forthcoming [tesseract_process_managers](https://github.com/ros-industrial-consortium/tesseract/pull/331) package in tesseract planning.

File: taskflow/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            CMAKE_CACHE_ARGS
+            -DCMAKE_INSTALL_PREFIX:STRING=${CMAKE_INSTALL_PREFIX}
+            -DCMAKE_BUILD_TYPE:STRING=Release
+            -DBUILD_TESTING=OFF
+            )
+
+        install(FILES package.xml DESTINATION share/Taskflow)
+    else()
+        message(WARNING "Taskflow not found and could not be installed due to CMake version < 3.9.0")
+    endif()
```
Comment (Levi-Armstrong) [inline taskflow/CMakeLists.txt:30]: need to add dummy install to make cmake happy as done below.

**Comment by Levi-Armstrong** [comment]
It is odd that it is wanting git user information.

**Comment by Levi-Armstrong** [comment]
Maybe try to add a before script to set the git user and email?

File: taskflow/CMakeLists.txt (hunk 1 above)
Comment (mpowelson) [reply taskflow/CMakeLists.txt:30]: Like this?

Comment (Levi-Armstrong) [reply taskflow/CMakeLists.txt:30]: Yep

**Comment by mpowelson** [comment]
Thanks for fixing that.

**Comment by Levi-Armstrong** [comment]
Also I just tested installing cmake using pip which gets you 3.18. I will update the docker for kinetic to install the latest version of cmake so taskflow can be used on 16.04.

**Comment by Levi-Armstrong** [comment]
I opened a [PR](https://github.com/ros-industrial/docker/pull/37) to address kinetic builds and git issue.

**Pull request merged**